common/src/main/java/io/netty/util/internal/logging/InternalLoggerFactory.java

```diff
@@ -77,7 +80,22 @@ InternalLogger newInstance(String name) {
 
     @SuppressWarnings("UnusedCatchParameter")
     private static InternalLoggerFactory newDefaultFactory(String name) {
-        InternalLoggerFactory f;
+        InternalLoggerFactory f = null;
+
+        String loggerType = SystemPropertyUtil.get("io.netty.logger.type");
+        if (loggerType != null) {
+            try {
+                Class<?> clazz = Class.forName(loggerType);
+                Field field = clazz.getField("INSTANCE");
+                f = (InternalLoggerFactory) field.get(null);
+                f.newInstance(name).debug("Using {} as the default logging framework", loggerType);
```

Comment:
This is making an assumption about the structure of the logger factory that might not be correct. If this is indeed accepted, it should at least be documented what the structure of an `InternalLoggerFactory` that can be set via a system property must be.